src/makeCalculator.js

```diff
@@ -4,7 +4,33 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  return {
+    result: 0,
+    add(digit) {
+      return (this.result += digit);
+    },
+    subtract(digit) {
+      return (this.result -= digit);
+    },
+    multiply(digit) {
+      return (this.result *= digit);
+    },
+    divide(digit) {
+      return (this.result /= digit);
+    },
+
+    operate(callback, number) {
+      callback.call(this, number);
+
+      return this;
+    },
```

Comment:
The `operate` method correctly uses the callback and provides chaining by returning `this`. Ensure that the callback is a function before calling it to avoid runtime errors.